src/Jenssegers/Mongodb/Connection.php

```diff
@@ -183,7 +183,7 @@ protected function getDsn(array $config)
             }
         }
 
-        return "mongodb://" . implode(',', $hosts) . "/{$database}";
+        return "mongodb://" . implode(',', $hosts);
```

Comment:
A agree with your reasons  :) but it may break many older project relying on this type of connection if they just upgrade!
So why not using a backward compatible trick like this : 

```php
return "mongodb://" . implode(',', $hosts) . ($database?"/{$database}":'');